crates/e2e/tests/e2e/replace_order.rs

```diff
@@ -23,6 +23,135 @@ async fn local_node_try_replace_someone_else_order() {
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        token_b.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        onchain.contracts().uniswap_v2_router.add_liquidity(
+            token_a.address(),
+            token_b.address(),
+            to_wei(1000),
+            to_wei(1000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Approve GPv2 for trading
+    tx!(
+        trader.account(),
+        token_a.approve(onchain.contracts().allowance, to_wei(15))
+    );
+
+    // Place Orders
+    let services = Services::new(&onchain).await;
+    services.start_protocol(solver).await;
+
+    let order = OrderCreation {
+        sell_token: token_a.address(),
+        sell_amount: to_wei(10),
+        buy_token: token_b.address(),
+        buy_amount: to_wei(5),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+    let balance_before = token_a.balance_of(trader.address()).call().await.unwrap();
+    onchain.mint_block().await;
+    let order_id = services.create_order(&order).await.unwrap();
+
+    let app_data = format!(
+        r#"{{
+              "version":"1.1.0",
+                  "metadata":{{
+                      "replacedOrder":{{
+                          "uid":"{}"
+                      }},
+                      "customStuff": 20
+                  }}
+              }}"#,
+        order_id
+    );
+
+    tracing::info!("Waiting for the old order to be executed");
+    wait_for_condition(TIMEOUT, || async {
+        let balance_after = token_a.balance_of(trader.address()).call().await.unwrap();
+        balance_before.saturating_sub(balance_after) == to_wei(10)
+    })
+    .await
+    .unwrap();
+
+    // Replace order
+    let new_order = OrderCreation {
+        sell_token: token_a.address(),
+        sell_amount: to_wei(3),
+        buy_token: token_b.address(),
+        buy_amount: to_wei(1),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        partially_fillable: false,
+        app_data: OrderCreationAppData::Full {
+            full: app_data.clone(),
+        },
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+    let response = services.create_order(&new_order).await;
+    let (error_code, _) = response.err().unwrap();
+    assert_eq!(error_code, StatusCode::BAD_REQUEST);
```

Comment:
This should assert that we are running into the replacement error and not perhaps some generic error.